### PR TITLE
Allow PHP unit tests to run in forked repos.

### PR DESCRIPTION
**Pull request by tebb**
This upgrades the workflow template file for new GitHub workflows to use PHP 7.4 and also to allow tests to be run in forked repos.

It is identical to that already in use for localgovdrupal/localgov.

### CODE REVIEW

**Comment by tebb** [comment]
I've realised the template itself was already configured to use PHP7.4, so this pr actually only adds forked repo testing to the template.

Note however that existing 'instances' i.e. test.yml files made from earlier versions of this template (before PHP 7.4 was specified) are in use for some LocalGov Drupal repos.  Those workflows may fail if PHP 8.0 is used from GitHub's ubuntu-latest image.  This issue https://github.com/localgovdrupal/localgov/issues/176 should rectify that.

**Review by paulpopus — APPROVED** [review]
I've tested this in the [localgov_theme repo](https://github.com/localgovdrupal/localgov_theme/issues/154
) and in the [localgov_core repo](https://github.com/localgovdrupal/localgov_core/pull/51), looks good!

Only failures were tests specific to particular modules.

File: workflow-templates/localgov-drupal-ci.yml [hunk 1]
```diff
@@ -26,21 +26,25 @@ jobs:
...
-      - name: Extract Git branch name outside of a pull request
+      - name: Save git branch and git repo names to env if not a pull request
         if: github.event_name != 'pull_request'
-        run: echo "GIT_BRANCH=${GITHUB_REF#refs/heads/}" >> $GITHUB_ENV
+        run: |
+          echo "GIT_BRANCH=${GITHUB_REF#refs/heads/}" >> $GITHUB_ENV
+          echo "GIT_REPO=${GITHUB_REPOSITORY}" >> $GITHUB_ENV
 
-      - name: Extract Git branch name from a pull request
+      - name: Save git branch and git repo names to env if is a pull request
```
Comment (Adnan-cds) [inline workflow-templates/localgov-drupal-ci.yml]: Err... there's a small typo :( @tebb

File: workflow-templates/localgov-drupal-ci.yml [hunk 2]
```diff
@@ -26,21 +26,25 @@ jobs:
...
+          echo "GIT_REPO=${{ github.event.pull_request.head.repo.full_name }}" >> $GITHUB_ENV
+
+      - name: Setup package source and authentication for the test target
+        run: |
+          composer --working-dir=html config repositories.1 vcs git@github.com:${GIT_REPO}.git
+          composer global config github-oauth.github.com ${{ github.token }}
 
-      - name: Grab test target
-        run: composer --working-dir=html require ${{ github.repository }}:"dev-${GIT_BRANCH} as 1.0.x-dev"
+      - name: Grab test target from this repo
```
Comment (Adnan-cds) [inline workflow-templates/localgov-drupal-ci.yml]: +1

File: workflow-templates/localgov-drupal-ci.yml (hunk 1 above)
Comment (tebb) [reply workflow-templates/localgov-drupal-ci.yml]: Thanks for reviewing Adnan.

Not quite sure what you want.  I was trying to keep it short and did mean 'is'???

I'll change to :  Save git branch and git repo names to environment if this is a pull request

... and change the similar, adjacent comment to match.

**Review by Adnan-cds — APPROVED** [review]
Looks good to me.  Thanks a lot :)

**Pull request merged**